pkg/config/config_template.yaml

```diff
@@ -1,424 +1,781 @@
+
+## @param log_file_max_rolls - integer - optional - default: 1
+## Maximum amount of "old" log files to keep.
+## Set to 0 to not limit the number of files to create.
 #
-# Duration of the restart interval in seconds
-# jmx_restart_interval: 5
+#  log_file_max_rolls: 1
+
+## @param log_to_syslog - boolean - optional - default: false
+## Set to 'true' to enable logging to syslog.
+## Note: Even if this option is set to 'false', the service launcher of your environment
+## may redirect the Agent process' stdout/stderr to syslog. In that case, if you wish
+## to disable logging to syslog entirely, set 'log_to_console' to 'false' as well.
 #
-# JMXFetch collects multiples instances concurrently. The following options may
-# help fine-tune the level of concurrency and timeouts that come into play during the
-# collection of metrics from configured instances:
+#  log_to_syslog: false
+
+## @param syslog_uri - string - optional
+## Define a custom remote syslog uri if needed. If 'syslog_uri' is left undefined/empty,
+## a local domain socket connection is attempted.
 #
-# Defines the maximum level of concurrency. Higher concurrency will increase CPU
-# utilization during metric collection. Lower concurrency will result in lower CPU
-# usage but may increase the total collection time - a value of 1 will process
-# instances serially. The total collection is allowed to take up to `jmx_collection_timeout`
-# seconds.
-# jmx_thread_pool_size: 3
+#  syslog_uri: <SYSLOG_URI>
+
+## @param syslog_rfc - boolean - optional - default: false
+## Set to 'true' to output in an RFC 5424-compliant format for Agent logs.
 #
-# Defines the maximum waiting period in seconds before timing up on metric collection.
-# jmx_collection_timeout: 60
+#  syslog_rfc: false
+
+## @param syslog_pem - string - optional
+## If TLS enabled, you must specify a path to a PEM certificate here.
 #
-# Defines the maximum level of concurrency. Higher concurrency will increase CPU
-# utilization during reconnection. Lower concurrency will result in lower CPU
-# usage but may increase the total reconnection time - a value of 1 will process
-# instance reconnections serially. In total, reconnections are allowed to take up to
-# `jmx_reconnection_timeout` seconds.
-# jmx_reconnection_thread_pool_size: 3
+#  syslog_pem: <PEM_CERTIFICATE_PATH>
+
+## @param syslog_key - string - optional
+## If TLS enabled, you must specify a path to a private key here.
 #
-# Determines the maximum waiting period in seconds before timing up on instance reconnection.
-# jmx_reconnection_timeout: 10
+#  syslog_key: <PEM_KEY_PATH>
+
+## @param syslog_tls_verify - boolean - optional - default: true
+## If TLS enabled, you may enforce TLS verification here.
 #
+#  syslog_tls_verify: true
+
 {{ end -}}
 {{- if .Autoconfig }}
-# Autoconfig
-#
-# Directory containing configuration templates
-# autoconf_template_dir: /datadog/check_configs
+
+####################################
+## Autoconfig Agent Configuration ##
```

Comment:
I believe that this should be renamed by `Container Autodiscovery Configuration` and should be moved before the agent logging instructions which are probably not used much (except for changing the logging level)